partner_risk_insurance/templates/invoice_risk_insurance_template.xml

```diff
@@ -0,0 +1,170 @@
+                </tr>
+            </t>
+            <t
+                t-if="move_first or move.partner_id.country_id != moves[move_index -1].partner_id.country_id"
+            >
+                <tr>
+                    <td colspan="5">
+                        <strong>
+                            <t t-out="move.partner_id.country_id.code" />
+                            -
+                            <t t-out="move.partner_id.country_id.name" />
+                        </strong>
+                    </td>
+                </tr>
+            </t>
+            <tr>
+                <td t-out="move.name" style="border: 1px solid black" />
+                <td t-out="move.partner_id.ref or ''" style="border: 1px solid black" />
+                <td t-out="move.partner_id.name" style="border: 1px solid black" />
+                <td
+                    t-out="move.amount_total"
+                    t-options='{"widget": "monetary", "display_currency": move.currency_id}'
+                    style="border: 1px solid black"
+                    align="right"
+                >
+                </td>
+                <td t-out="move.date" style="border: 1px solid black" align="right"/>
+                <t
+                    t-set="insurance_country_total"
+                    t-value="insurance_country_total + move.amount_total"
+                />
+            </tr>
+            <t
+                t-if="move_last or move.partner_id.country_id != moves[move_index + 1].partner_id.country_id"
+            >
+                <tr>
+                    <td colspan="5" align="right">
+                        <strong >
+                            Total by country:
+                            <t
+                                t-out="insurance_country_total"
+                                t-options='{"widget": "monetary", "display_currency": move.currency_id}'
+                            >
+                            </t>
+                        </strong>
+                    </td>
+                </tr>
+                <t t-set="insurance_country_total" t-value="0" />
+            </t>
+        </t>
+        <tr>
+            <td colspan="5" align="right">
+                <strong>
+                    Total
+                    <t t-if="is_insured">insured</t>
+                    <t t-else="">not insured</t>
+                    customers:
+                    <t
+                        t-out="insurance_partial_total"
+                        t-options='{"widget": "monetary", "display_currency": move.currency_id}'
+                        align="right"
+                    >
+                    </t>
+                </strong>
+            </td>
+        </tr>
+    </template>
+    <template id="report_invoice_risk_insurance_document">
+        <t t-call="web.external_layout">
+
+            <t
+                t-set="insurance_total_amount"
+                t-value="sum(docs.mapped('amount_total'))"
```

Comment:
Why do you need to go through all the records again to calculate the total instead of accumulating it in the same way as you accumulate the total per country?

Reply:
The problem is that I can't accumulate the total because it's value is not stored between the two templates that I am using to render the report. I changed how is the partial calculated, but as I said for now I don't know how this value could be preserved.